contracts/rust/deployer/src/builder.rs

```diff
@@ -363,6 +478,122 @@ impl<P: Provider + WalletProvider> DeployerArgs<P> {
+        let timelock_operation_type = self
+            .timelock_operation_type
+            .context("Timelock operation type not found")?;
+        let target_contract = self
+            .timelock_target_contract
+            .clone()
+            .context("Timelock target not found")?;
+        let value = self
+            .timelock_operation_value
+            .context("Timelock operation value not found")?;
+        let function_signature = self
+            .timelock_operation_function_signature
+            .as_ref()
+            .context("Timelock operation function signature not found")?;
+        let function_values = self
+            .timelock_operation_function_values
+            .clone()
+            .context("Timelock operation function values not found")?;
+        let salt = self
+            .timelock_operation_salt
+            .clone()
+            .context("Timelock operation salt not found")?;
+        let delay = self
+            .timelock_operation_delay
+            .context("Timelock operation delay not found")?;
+
+        let (target_addr, contract_type) = match target_contract.as_str() {
+            "FeeContract" => (
+                contracts
+                    .address(Contract::FeeContractProxy)
+                    .context("FeeContractProxy address not found")?,
+                Contract::FeeContractProxy,
+            ),
+            "EspToken" => (
+                contracts
+                    .address(Contract::EspTokenProxy)
+                    .context("EspTokenProxy address not found")?,
+                Contract::EspTokenProxy,
+            ),
+            "LightClient" => (
+                contracts
+                    .address(Contract::LightClientProxy)
+                    .context("LightClientProxy address not found")?,
+                Contract::LightClientProxy,
+            ),
+            "StakeTable" => (
+                contracts
+                    .address(Contract::StakeTableProxy)
+                    .context("StakeTableProxy address not found")?,
+                Contract::StakeTableProxy,
+            ),
+            _ => anyhow::bail!("Invalid target contract: {}", target_contract),
+        };
+
+        let function_calldata = encode_function_call(function_signature, function_values.clone())
+            .context("Failed to encode function data")?;
+
+        // Parse salt from string to B256
+        let salt_bytes = if salt == "0x" || salt.is_empty() {
+            B256::ZERO // Use zero salt if empty
+        } else if let Some(stripped) = salt.strip_prefix("0x") {
+            B256::from_hex(stripped).context("Invalid salt hex format")?
+        } else {
+            B256::from_hex(&salt).context("Invalid salt hex format")?
+        };
+
+        let timelock_operation_data = TimelockOperationData {
+            target: target_addr,
+            value,
+            data: function_calldata,
+            predecessor: B256::ZERO, // Default to no predecessor
```

Comment:
The `predecessor` is for chaining operations, or?

Reply:
yes for chaining operations